### PR TITLE
Feature 29 refine timeentry

### LINKED ISSUE #29: Refine TimeEntry
- Rename TimeEntry to WorkEvent
- Remove endTime field

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #29

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/vp/usermanagement/messaging/MessagingController.kt [hunk 1]
```diff
@@ -40,30 +38,19 @@ class MessagingController: WithCoroutineScope() {
...
-            } else {
-                logger.debug("No incomplete valid time entries found for driver ${foundDriver.id}")
-            }
 
-            timeEntryController.create(
+            workEventController.create(
                 employee = foundDriver,
-                startTime = parsed.time,
-                workEventType = parsed.workEventType
+                startTime = event.time,
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/vp/usermanagement/messaging/MessagingController.kt:52]: Why startTime?

File: src/main/kotlin/fi/metatavu/vp/usermanagement/rest/AbstractApi.kt [hunk 2]
```diff
@@ -179,6 +179,10 @@ abstract class AbstractApi: WithCoroutineScope() {
         return createError(Response.Status.NOT_FOUND, message)
     }
 
+    protected fun createNotFoundWithMessage(entity: String, id: UUID): Response {
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/vp/usermanagement/rest/AbstractApi.kt:182]: Docs

File: src/main/resources/db/changeLog.xml [hunk 3]
```diff
@@ -75,7 +75,24 @@
...
+    <changeSet id="replace_timeentry_with_workevent" author="Ville Juutila">
+        <dropTable tableName="timeentry"/>
+        <createTable tableName="workevent">
+            <column name="id" type="binary(16)">
+                <constraints nullable="false" primaryKey="true"/>
+            </column>
+            <column name="employeeId" type="binary(16)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="startTime" type="datetime(6)">
```
Comment (anttileppa) [inline src/main/resources/db/changeLog.xml:90]: Why this is startTime, not just time?

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/vp/usermanagement/messaging/MessagingController.kt (hunk 1 above)
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/vp/usermanagement/messaging/MessagingController.kt]: Good point. Renamed to just time.

File: src/main/resources/db/changeLog.xml (hunk 3 above)
Comment (villejuutila) [reply src/main/resources/db/changeLog.xml]: Good point. Renamed to just time.

File: src/main/kotlin/fi/metatavu/vp/usermanagement/rest/AbstractApi.kt (hunk 2 above)
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/vp/usermanagement/rest/AbstractApi.kt:189]: Resolved

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**